### PR TITLE
Emformer augmented with positional encoding and convolution module

### PR DESCRIPTION
**Pull request by yaozengwei**
The training and decoding codes are based on `pruned_transducer_stateless4`, with scaling mechanism and averaged model saving.

Now this version supports `positional encoding` for emformer attention and also adds `convolution module`.

The emformer implementation is based on the torchaudio released codes:
- emformer attention: https://github.com/pytorch/audio/blob/main/torchaudio/models/emformer.py (without positional encoding yet)
- convolution module: https://github.com/pytorch/audio/blob/main/torchaudio/prototype/models/conv_emformer.py

### CODE REVIEW

**Comment by yaozengwei** [comment]
The `test_emformer.py` provides test functions to check output shapes of modules, and also the consistency between training with mask simulation and streaming chunk-wise inference.

**Comment by danpovey** [comment]
Thanks!  Will try to have a look tomorrow

**Comment by yaozengwei** [comment]
This is a training script example: 
```
export CUDA_VISIBLE_DEVICES="0,1,2,3,4,5,6,7"

./conv_emformer_transducer_stateless/train.py \                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           
  --world-size 8 \                                                                                                                                                          
  --num-epochs 30 \                                                                                                                                                         
  --start-epoch 1 \                                                                                                                                                         
  --exp-dir conv_emformer_transducer_stateless/exp-960_num_layer-12_chunk-32_conv-31_left-32_right-8_memory-32 \                                                            
  --full-libri 1 \                                                                                                                                                          
  --max-duration 300 \                                                                                                                                                      
  --master-port 12321 \                                                                                                                                                     
  --num-encoder-layers 12 \                                                                                                                                                 
  --chunk-length 32 \                                                                                                                                                       
  --cnn-module-kernel 31 \
  --left-context-length 32 \
  --right-context-length 8 \
  --memory-size 32
```
